pkg/repo/callrepo.go

```diff
@@ -79,3 +87,71 @@ func NewCallsRepository(cfg *CallsConfig) (*CallsRepository, error) {
 func (r *CallsRepository) GetCall(method string) core.Handler {
 	return r.calls[method]
 }
+
+// topSortDFS performs a topological sort on a slice of BackendConfig using Depth-First Search (DFS).
+// It takes a slice of BackendConfig as input and returns a sorted slice of BackendConfig and an error.
+// It returns an error if a circular dependency is detected among the BackendConfig elements.
+// Each BackendConfig element must have a unique ResponseBody and a list of dependencies specified in DependsOn.
+func topSortDFS(be []BackendConfig) ([]BackendConfig, error) {
+	graph := createDepGraph(be)
+	visited := make(map[string]bool)
+	recStack := make(map[string]bool)
+
+	var sorted []BackendConfig
+
+	var dfs func(string) error
+	dfs = func(v string) error {
+		if recStack[v] {
+			return fmt.Errorf("circular dependency detected")
+		}
+
+		if visited[v] {
+			return nil
+		}
+
+		visited[v] = true
+		recStack[v] = true
+
+		for _, u := range graph[v] {
+			if err := dfs(u); err != nil {
+				return err
+			}
+		}
```

Comment:
Here are my suggestions to this function

1. use map for quick lookup
2. some more error messages
3. use defer for cleanup

```suggestion
// topSortDFS performs a topological sort on a slice of BackendConfig using Depth-First Search (DFS).
// It takes a slice of BackendConfig as input and returns a sorted slice of BackendConfig and an error.
// It returns an error if a circular dependency is detected among the BackendConfig elements.
// Each BackendConfig element must have a unique ResponseBody and a list of dependencies specified in DependsOn.
func topSortDFS(be []BackendConfig) ([]BackendConfig, error) {
	graph := createDepGraph(be)
	visited := make(map[string]bool)
	recStack := make(map[string]bool)
	sorted := []BackendConfig{}

	// Create a map for quick index lookup
	indexMap := make(map[string]int)
	for i, b := range be {
		indexMap[b.ResponseBody] = i
	}

	var dfs func(string) error
	dfs = func(v string) error {
		if recStack[v] {
			return fmt.Errorf("circular dependency detected at %s", v)
		}
		if visited[v] {
			return nil
		}

		visited[v] = true
		recStack[v] = true
		defer func() { recStack[v] = false }() // Ensure recStack is reset

		for _, u := range graph[v] {
			if err := dfs(u); err != nil {
				return err
			}
		}

		sorted = append(sorted, be[indexMap[v]])
		return nil
	}

	for k := range graph {
		if err := dfs(k); err != nil {
			return nil, err
		}
	}

	return sorted, nil
}
```